### ISSUE TITLE
[data grid] Row reordering triggers cell selection process

### ISSUE THREAD

**Issue by PEsteves8** [mui/mui-x#11753]
### Steps to reproduce

Link to live example: https://mui.com/x/react-data-grid/clipboard/ (just add rowReordering to the paste example

Steps:
1. Drag a row
2. Once you finish , move the cursor to the middle of the table
3. You'll have a grid of highlighted cells for selection

### Current behavior

Finishing a row reordering starts cell selection

### Expected behavior

Row reordering shouldn't keep a cell selection mode

Perhaps drag columns shouldn't be selectable, is it possible to disable it from the api?

**Search keywords**: row reordering cell selection
**Order ID**: 82566

**Comment by michelengelen** [comment]
Hey @PEsteves8 and thanks For raising this issue.
I can confirm that it is indeed a bug.

Here is a recording of it. Notice that after the first drag and drop the cell selection event is not cancelled.

https://github.com/mui/mui-x/assets/32863416/9d17f2fe-85c5-4f6f-9c04-3d0996f029b5

The fix for this can be pretty straight forward. Here is a diff to get started:

```diff
diff --git a/packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts b/packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts
index 7d3d888d5..5877bfe51 100644
--- a/packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts
+++ b/packages/grid/x-data-grid-premium/src/hooks/features/cellSelection/useGridCellSelection.ts
@@ -22,6 +22,7 @@ import {
   gridClasses,
   gridFocusCellSelector,
   GridCellParams,
+  GRID_REORDER_COL_DEF,
 } from '@mui/x-data-grid-pro';
 import { gridCellSelectionStateSelector } from './gridCellSelectionSelector';
 import { GridCellSelectionApi } from './gridCellSelectionInterfaces';
@@ -221,6 +222,10 @@ export const useGridCellSelection = (
         return;
       }

+      if (params.field === GRID_REORDER_COL_DEF.field) {
+        return;
+      }
+
       const focusedCell = gridFocusCellSelector(apiRef);
       if (hasClickedValidCellForRangeSelection(params) && event.shiftKey && focusedCell) {
         event.preventDefault();
```

This will prevent the cell selection to start when clicking the reorder cell.

**Comment by PEsteves8** [comment]
Nice, thanks. I created a PR : https://github.com/mui/mui-x/pull/11783

Hopefully I followed the instructions correctly. Although there was a reference for signing a CLA that I didn't quite get. Maybe it'll come up in the discussion. 

(Also, just a note, I got a bit tripped up following the instructions from material-ui when it indicated that I had to use pnpm. It took me a while to realize that mui-x is using yarn)
